Corrfunc/theory/DD.py

```diff
@@ -14,7 +14,7 @@
 
 
 def DD(autocorr, nthreads, binfile, X1, Y1, Z1, weights1=None, periodic=True,
-       X2=None, Y2=None, Z2=None, weights2=None, verbose=False, boxsize=0.0,
+       X2=None, Y2=None, Z2=None, weights2=None, verbose=False, boxsize=None,
```

Comment:
Since `boxsize` is now tied to `perioodic`,  we should move the `boxsize` parameter to follow right after `periodic`. Imagine the case with `autocorr=True` and `periodic=True`, the computation is still not specified until `boxsize` is specified -> `boxsize` should now appear before `X2`